File: test/language/expressions/cpp/SumOperatorTest.cpp
#include "math.h"

#include "gtest/gtest.h"

#include "expressions/sum_operator/SumFunction.h"

namespace expressions
{
namespace sum_operator
{

TEST(SumOperatorTest, GetSumFixedArray)
{
    SumFunction sumFunction;
    zserio::UInt8Array fixedArray;
    uint16_t expectedSum = 0;
    for (uint8_t i = 1; i <= 10; ++i)
    {
        fixedArray.push_back(i);
        expectedSum += i;
    }
    sumFunction.setFixedArray(fixedArray);
    ASSERT_EQ(expectedSum, sumFunction.getSumFixedArray());
}

} // namespace sum_operator
} // namespace expressions

File: test/language/expressions/java/expressions/SumOperatorTest.java
package expressions;

import static org.junit.Assert.*;

import org.junit.Test;

import zserio.runtime.array.UnsignedByteArray;

import expressions.sum_operator.SumFunction;

public class SumOperatorTest
{
    @Test
    public void getSumFixedArray()
    {
        final SumFunction sumFunction = new SumFunction();
        final short[] fixedArrayData = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
        final UnsignedByteArray fixedArray = new UnsignedByteArray(fixedArrayData, 0, fixedArrayData.length);
        sumFunction.setFixedArray(fixedArray);

        int expectedSum = 0;
        for (int element : fixedArrayData)
            expectedSum += element;
        assertEquals(expectedSum, sumFunction.funcGetSumFixedArray());
    }
}
